src/main/java/io/github/minecraftcursedlegacy/mixin/MixinTessellator.java

```diff
@@ -0,0 +1,55 @@
+package io.github.minecraftcursedlegacy.mixin;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import io.github.minecraftcursedlegacy.impl.client.TextureRegistryImpl;
+import net.minecraft.client.render.Tessellator;
+
+@Mixin(Tessellator.class)
+public class MixinTessellator {
+	@Inject(at = @At("HEAD"), method = "vertex", cancellable = true)
+	public void vertex(double d, double d1, double d2, double d3, double d4, CallbackInfo bruh) {
+		if (TextureRegistryImpl.vertexHacksEnabled) {
+			Double texture1 = null;
+			Double texture2 = null;
+
+			int var4 = TextureRegistryImpl.var4hack;
+			float var5 = ((float)(var4 % 16 * 16) + 0.0F) / 256.0F;
+			float var6 = ((float)(var4 % 16 * 16) + 15.99F) / 256.0F;
+			float var6_2 = ((float)(var4 % 16 * 16) + 16F) / 256.0F;
+			float var7 = ((float)(var4 / 16 * 16) + 0.0F) / 256.0F;
+			float var8 = ((float)(var4 / 16 * 16) + 15.99F) / 256.0F;
```

Comment:
Maybe this method could do with better var naming?

It looks like you've done something very smart here but I have no clue what :p

Reply:
It's all the cloned logic from where vanilla renders items, I'm not sure this is the most sensible way to be doing this given `vertex` is called a lot. It would be better to change the texture position directly in the rendering methods in `class_556` and `ItemRenderer`

Reply:
That would require at least 10 times the code.